### PR TITLE
Some improvements on Settings controller

### PR DESCRIPTION
**Pull request by billmn**
- Redirect back instead of always redirect to setup page
- Use URL generation by route instead of specify it manually
- More expressive image check

### CODE REVIEW

**Comment by jbrooksuk** [comment]
- We actually avoid doing `Redirect::back` purposefully because it's insecure. A MITM attack could redirect you somewhere else.

**Comment by billmn** [comment]
What about this alternative?

**Comment by jbrooksuk** [comment]
Still open to attack. Hmm, maybe we could store a session value with the last route when in the dashboard? @GrahamCampbell?

**Comment by billmn** [comment]
This require to reflash session on reload. Anyway I think is not the right thing to redirect always to setup section

**Comment by billmn** [comment]
@jbrooksuk I've changed the code to use session like your suggestion. Take a look ;-)

File: app/Http/Controllers/Dashboard/SettingsController.php [hunk 1]
```diff
@@ -68,6 +68,8 @@ public function showSetupView()
     {
         $this->subMenu['setup']['active'] = true;
 
+        session()->flash('redirect_to', $this->subMenu['setup']['url']);
```
Comment (jbrooksuk) [inline app/Http/Controllers/Dashboard/SettingsController.php]: @GrahamCampbell what do you think, should we be using the Session facade here?

Comment (GrahamCampbell) [inline app/Http/Controllers/Dashboard/SettingsController.php]: meh, doesn't really matter - if you want to, go for it ;)

Comment (GrahamCampbell) [inline app/Http/Controllers/Dashboard/SettingsController.php]: nice to be consistent with what ever we use else where

Comment (billmn) [inline app/Http/Controllers/Dashboard/SettingsController.php]: @GrahamCampbell make sense. I'll use facade.

**Comment by billmn** [comment]
Now with facade

**Comment by jbrooksuk** [comment]
Squash into one commit and we'll get it merged! :)

**Comment by GrahamCampbell** [comment]
> Squash into one commit and we'll get it merged! :)

I can do that. :)

**Comment by GrahamCampbell** [comment]
Squashed and merged. Thank you @billmn.

**Comment by GrahamCampbell** [comment]
Missed the `#` in the commit so this PR didn't auto-close. Oops...

**Pull request closed without merging**